core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -1561,4 +1561,18 @@ class ParserTest extends munit.ScalaCheckSuite {
     }
   }
 
+  property("p.filter(_ => true) == p") {
+    forAll(ParserGen.gen, Arbitrary.arbitrary[String]) { (genP, str) =>
+      val res0 = genP.fa.filter(_ => true).parse(str)
+      val res1 = genP.fa.parse(str)
+      assertEquals(res0, res1)
+    }
+  }
+
```

Comment:
Another interesting pair of laws: if we pa.ap(pb) should be the same as select(pa.map(Left), pb, fail)

Similarly on the right hand side.